framework/OpenMod.Core/Users/UserTypeConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using OpenMod.API.Users;
+using OpenMod.Core.Helpers;
+
+namespace OpenMod.Core.Users
+{
+    /// <summary>
+    /// Converts <i>actorType:actorNameOrId</i> (for example <i>player:Foo Bar</i>)
+    /// to <see cref="IUser"/> and vice versa.
+    /// The <see cref="UserTypeConverter.DefaultActorType"/> will be used if none is specified.
+    /// </summary>
+    public sealed class UserTypeConverter : TypeConverter
+    {
+        public string DefaultActorType;
+        public char Separator;
+
+        public UserTypeConverter(string defaultActorType = KnownActorTypes.Player, char separator = ':')
+        {
+            DefaultActorType = defaultActorType;
+            Separator = separator;
+        }
+
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            return sourceType == typeof(string) || base.CanConvertFrom(context, sourceType);
+        }
+
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            if (!(value is string arg))
+                return base.ConvertFrom(context, culture, value);
+
+            IUserManager userManager = context.GetRequiredService<IUserManager>();
```

Comment:
This will not work because the DI Container / ServiceProvider is never passed to this converter

Reply:
It is passed here
https://github.com/openmod/openmod/blob/405876f4c975da3e4360dde9013d0d363dfdb8c8/framework/OpenMod.Core/Helpers/TypeConverterHelper.cs#L21-L24